### DISCUSSION TITLE
Change of location of a spool is not visible in home assistant

### DISCUSSION THREAD

**Discussion by tsgwiro1** [Disane87/spoolman-homeassistant#144] [Q&A] [answered]
Hi all

I have all my active / open spools in seperate dry boxes. An active spool has its dry box where it is stored as location set (e.g. "Dry Box 1", "Dry Box 2", ...). All new and unopened spools are in one additional location called "Lager". As soon as a new spool from "Lager" is used for printing, its location is changed from "Lager" to whatever "Dry Box n" it is assigned to. The location change is done directly in the spoolman UI. Unfortunately this change is not visible in home assistant, the spools location does not change.

Is there anything that I'm doing wrong? How can I trigger an update of the spools locations in HA?

**Comment by Disane87** [comment]
You have to reload the entire integration atm to achieve this. This  is currently a bit odd. due to the nature HA creates the areas. At the moment I'm planning to refactor this and remove the ability to have all spools grouped by an area. In the future I'm planning to use tags or something else to prevent this.

**Reply by tsgwiro1** [reply]
Thanks a lot, that worked :-)

**Reply by rsocko** [reply]
I was wondering the same thing. The reload certainly helps. 

In Spoolman I currently keep a location of AMS and am moving spools in and out of the AMS location. It seems as though when I move a spool from one location to another in Spoolman, there is often a leftover entity. Home Assistant reports: 

`This entity is no longer being provided by the spoolman integration. If the entity is no longer in use, delete it in settings.`

The spool that was moved to a new location shows correctly as an entity in that new location. But it seems to have a 'ghost' entity still in the original location that Spoolman is no longer providing.

Is there a way to automatically clean this up? Or curious if you have made progress or thought further about how you are going to manage the location information?

**Reply by Disane87** [reply]
It's possible that this issue is fixed by @bplainia in the latest version. Would be nice if you could test it.

**Reply by rsocko** [reply, marked as answer]
I've been having the same issue. When I upgraded to 1.0 it didn't seem to correct the issue.

However, earlier today I deleted the integration, re-downloaded again from HACS (lastest version 1.0) and re-added the integration.

I can report that it appears to be working perfectly at this point. I have moved several spools from different locations in Spoolman and moved them into my "AMS" location and others out of "AMS" and it doesn't appear to be leaving any 'orphan' entities behind.

I did accept all of the integration defaults when adding - which means I now have a Home Assistant Area for each of the locations in Spoolman. I would prefer not to have those extra Areas confusing the rest of my home assistant setup. So I will try removing the Area or setting them all to my office (where the printer and spools are all located in my house) and continue to monitor to verify if the fix is still working.
